spec/requests/api/v3/authentication_spec.rb

```diff
@@ -0,0 +1,205 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+
+describe API::V3, type: :request do
+  describe 'basic auth' do
+    let(:user)     { FactoryGirl.create :user }
+    let(:resource) { "/api/v3/users/#{user.id}" }
+
+    let(:response_401) do
+      {
+        '_embedded'       => {},
+        '_type'           => 'Error',
+        'errorIdentifier' => 'urn:openproject-org:api:v3:errors:MissingPermission',
+        'message'         => 'You need to be authenticated to access this resource.'
+      }
+    end
+
+    Strategies = OpenProject::Authentication::Strategies::Warden
+
+    def basic_auth(user, password)
+      credentials = ActionController::HttpAuthentication::Basic.encode_credentials user, password
+      { 'HTTP_AUTHORIZATION' => credentials }
+    end
+
+    shared_examples 'it is basic auth protected' do
+      context 'without credentials' do
+        before do
+          get resource
+        end
+
+        it 'should return 401 unauthorized' do
+          expect(response.status).to eq 401
+        end
+
+        it 'should return the correct JSON response' do
+          expect(JSON.parse(response.body)).to eq response_401
+        end
```

Comment:
This + the 401 check should be replaceable with `it_behaves_like 'unauthenticated access'`.

The afforementioned shared example can be found in `spec/requests/api/v3/support/response_examples.rb` and uses `last_response` which is hopefully equivalent to `response` ^^